paywall/src/components/lock/ConfirmingKeyLock.js

```diff
@@ -14,7 +14,7 @@ import BalanceProvider from '../helpers/BalanceProvider'
 import withConfig from '../../utils/withConfig'
 
 export const ConfirmingKeyLock = ({ lock, transaction, config }) => (
-  <LockWrapper id={`Confirming_${lock.address}`}>
+  <LockWrapper lock={lock}>
```

Comment:
I think that rather than using `id` which are adding too much coupling we can use a `data-address` attribute to identify locks based on their addresses. The `LockWrapper` component does that.